jbmc/src/java_bytecode/java_bytecode_instrument.cpp

```diff
@@ -592,6 +593,30 @@ void java_bytecode_instrument_symbol(
   instrument(to_code(symbol.value));
 }
 
+/// Instruments the start function with an assertion that checks whether
+/// an exception has escaped the entry point
+/// \param symbol_table: global symbol table
+void java_bytecode_instrument_uncaught_exceptions(symbol_tablet &symbol_table)
+{
+  // retrieve the exception variable
+  const symbolt &exc_symbol =
+    symbol_table.lookup_ref(JAVA_ENTRY_POINT_EXCEPTION_SYMBOL);
+
+  symbolt &function =
+    symbol_table.get_writeable_ref(goto_functionst::entry_point());
```

Comment:
Huh? Why not just do this in `java_entry_point`, which already builds the machinery to catch the toplevel exception and record it as an `OUTPUT`.

Reply:
See comment by @thk123 above for keeping the generation of instrumentations in the same place.

Reply:
hmm. This seems very brittle at least, as it will break if the structure of __CPROVER_start changes incidentally. How about `java_entry_point` introduces a hook, `on_uncaught_toplevel_exception()` or something, which instrument populates?

Reply:
I've added some invariants to check that the entry point looks sane. Adding a separate seemed an overkill.